### PR TITLE
KAFKA-16487: Support to define server properties by ClusterTestDefaults

### PR DESCRIPTION
**Pull request by FrankYang0529**
Sometimes we want to define server properties for all test cases, and using `BeforeEach` to modify the `ClusterConfig` is the only way. The side effect is that the IDE does not like the style since IDE can't recognize custom ParameterResolver of `ClusterConfig`.

The alternative is that we can take `ClusterInstance` from constructor first, and then we modify the inner `ClusterConfig` in `BeforeEach` phase. However, that may confuse users about the life cycle of "configs".

In short, I prefer to define the server property by `ClusterTestDefaults`. It already includes some server-side default property, and we can enhance that to deal with more existent test case.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/test/java/kafka/test/annotation/ClusterTestDefaults.java [hunk 1]
```diff
@@ -39,4 +39,5 @@
     int brokers() default 1;
     int controllers() default 1;
     boolean autoStart() default true;
+    ClusterConfigProperty[] serverProperties() default {};
```
Comment (chia7712) [inline core/src/test/java/kafka/test/annotation/ClusterTestDefaults.java:43]: Could you add comments for this new fields?

**Review by chia7712 — COMMENTED** [review]
@FrankYang0529 thanks for this improvement. just one small comment

File: core/src/test/java/kafka/test/annotation/ClusterTestDefaults.java (hunk 1 above)
Comment (FrankYang0529) [reply core/src/test/java/kafka/test/annotation/ClusterTestDefaults.java:43]: Yes, updated it. Thank you.

File: core/src/test/java/kafka/test/junit/ClusterTestExtensions.java [hunk 2]
```diff
@@ -190,6 +190,9 @@ private void processClusterTest(ExtensionContext context, ClusterTest annot, Clu
         }
 
         Properties properties = new Properties();
+        for (ClusterConfigProperty property : defaults.serverProperties()) {
```
Comment (chia7712) [inline core/src/test/java/kafka/test/junit/ClusterTestExtensions.java:192]: this is unrelated to this PR, but could we update `config.serverProperties` instead of creating a temporary `properties`?

**Review by chia7712 — COMMENTED** [review]
@FrankYang0529 sorry that I leave a refactor-related comment. please take a look.

File: core/src/test/java/kafka/test/ClusterTestExtensionsTest.java [hunk 3]
```diff
@@ -89,20 +91,24 @@ public void testClusterTemplate() {
...
-            @ClusterConfigProperty(key = "spam", value = "eggz")
+            @ClusterConfigProperty(key = "spam", value = "eggz"),
+            @ClusterConfigProperty(key = "default.key", value = "overwrite.value")
         })
     })
     public void testClusterTests() {
         if (clusterInstance.clusterType().equals(ClusterInstance.ClusterType.ZK)) {
             Assertions.assertEquals(clusterInstance.config().serverProperties().getProperty("foo"), "bar");
             Assertions.assertEquals(clusterInstance.config().serverProperties().getProperty("spam"), "eggs");
+            Assertions.assertEquals(clusterInstance.config().serverProperties().getProperty("default.key"), "default.value");
```
Comment (chia7712) [inline core/src/test/java/kafka/test/ClusterTestExtensionsTest.java]: the first argument is "expected", and so `default.value` should be the first input.

File: core/src/test/java/kafka/test/junit/ClusterTestExtensions.java (hunk 2 above)
Comment (FrankYang0529) [reply core/src/test/java/kafka/test/junit/ClusterTestExtensions.java:192]: Updated it. Thank you.

File: core/src/test/java/kafka/test/ClusterTestExtensionsTest.java (hunk 3 above)
Comment (FrankYang0529) [reply core/src/test/java/kafka/test/ClusterTestExtensionsTest.java]: Changed the input order. Thank you.

**Review by chia7712 — APPROVED** [review]
LGTM
